src/hooks/useAxios.jsx

```diff
@@ -0,0 +1,23 @@
+import { useState } from "react";
+import axios from "axios";
+
+export default function useAxios() {
+  const [response, setResponse] = useState([]);
+  const [err, setErr] = useState([]);
+  const asyncFn = async ({ url, body, header, method }) => {
+    try {
+      if (method === "post") {
+        const response = await axios.post(url, body, { headers: header });
+        setResponse(response);
+      } else if (method === "get") {
+        const response = await axios.get(url, { headers: header });
+        console.log("await", response);
+        setResponse(response);
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
```

Comment:
Please let me know if this works. 

i moved away from useAxios after I figured out that all my api calls are inside functions within context

Reply:
i tried this but did not seem to work, but kept it to revisit when time permits. will let you know how it goes